contrib/platform/test/com/sun/jna/platform/win32/Shell32Test.java

```diff
@@ -131,16 +135,89 @@ public void testResizeDesktopFromTop() throws InterruptedException {
 
     }
 
-    public void testSHGetKnownFolderPath()
-    {
-        int flags = ShlObj.KNOWN_FOLDER_FLAG.NONE.getFlag();
-        PointerByReference outPath = new PointerByReference();
-        HANDLE token = null;
-        GUID guid = KnownFolders.FOLDERID_Fonts;
-        HRESULT hr = Shell32.INSTANCE.SHGetKnownFolderPath(guid, flags, token, outPath);
-
-        Ole32.INSTANCE.CoTaskMemFree(outPath.getValue());
-
-        assertTrue(W32Errors.SUCCEEDED(hr.intValue()));
-    }
+	public void testSHGetKnownFolderPath() {
+		int flags = ShlObj.KNOWN_FOLDER_FLAG.NONE.getFlag();
+		PointerByReference outPath = new PointerByReference();
+		HANDLE token = null;
+		GUID guid = KnownFolders.FOLDERID_Fonts;
+		HRESULT hr = Shell32.INSTANCE.SHGetKnownFolderPath(guid, flags, token, outPath);
+
+		Ole32.INSTANCE.CoTaskMemFree(outPath.getValue());
+
+		assertTrue(W32Errors.SUCCEEDED(hr.intValue()));
+	}
+
+	public void testSHEmptyRecycleBin() {
+		File file = new File(System.getProperty("java.io.tmpdir") + System.nanoTime() + ".txt");
+		try {
```

Comment:
You are relying on the fact the _System.getProperty("java.io.tmpdir")_ returns a path terminated with backslash. Better to check if this is indeed the case and append _File.separator_ if not.